cfn-resources/global-cluster-config/cmd/resource/resource.go

```diff
@@ -0,0 +1,365 @@
+		if err := removeManagedNamespaces(context.Background(), client, remove, projectID, clusterName); err != nil {
+			_, _ = logger.Debugln("Error -Unable to remove Namespace")
+		}
+	}
+	if currentModel.RemoveAllZoneMapping != nil && *currentModel.RemoveAllZoneMapping {
+		_, _, err := client.GlobalClusters.DeleteCustomZoneMappings(context.Background(), projectID, clusterName)
+		if err != nil {
+			return progressEvents.GetFailedEventByCode(fmt.Sprintf("Failed to custom zones : %s", err.Error()),
+				cloudformation.HandlerErrorCodeInvalidRequest), nil
+		}
+	}
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Delete Complete",
+	}, nil
+}
+
+func setup() {
+	util.SetupLogger("mongodb-atlas-global-cluster-config")
+}
+
+func isExist(client mongodbatlas.Client, currentModel *Model) bool {
+	setup()
+	config, _, err := ReadConfig(client, currentModel)
+	_, _ = logger.Debugf("%v", config)
+	if err == nil && config != nil {
+		return true
+	}
+	return false
+}
+
+// ValidateRequest function to validate the request
+func ValidateRequest(currentModel *Model) (handler.ProgressEvent, *mongodbatlas.Client, error) {
+	setup() // logger setup
+
+	_, _ = logger.Debugf("Request - snapshot restore starts currentModel:%+v", currentModel)
+
+	// Validate required fields in the request
+	if modelValidation := validateModel(RequiredFields, currentModel); modelValidation != nil {
+		return *modelValidation, nil, errors.New("required field not found")
+	}
+
+	// Create MongoDb Atlas Client using keys
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = logger.Warnf(constants.ErrorCreateMongoClient, err)
+		return progressEvents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
+			cloudformation.HandlerErrorCodeInvalidRequest), nil, err
+	}
+	return handler.ProgressEvent{}, client, nil
+}
+
+// function to validate inputs to all actions
+func validateModel(fields []string, model *Model) *handler.ProgressEvent {
+	return validator.ValidateModel(fields, model)
+}
+
+func removeManagedNamespaces(ctx context.Context, conn *mongodbatlas.Client, remove []ManagedNamespace, projectID, clusterName string) error {
+	for _, m := range remove {
+		addManagedNamespace := &mongodbatlas.ManagedNamespace{
+			Collection:     *m.Collection,
+			Db:             *m.Db,
+			CustomShardKey: *m.CustomShardKey,
+		}
+		addManagedNamespace.IsCustomShardKeyHashed = pointy.Bool(*m.IsCustomShardKeyHashed)
+		addManagedNamespace.IsShardKeyUnique = pointy.Bool(*m.IsShardKeyUnique)
+		_, _, err := conn.GlobalClusters.DeleteManagedNamespace(ctx, projectID, clusterName, addManagedNamespace)
+
+		if err != nil {
+			_, _ = logger.Warnf("error while removing namespace:%+v", err)
+		}
+	}
+
+	return nil
+}
+
+func addManagedNamespaces(ctx context.Context, conn *mongodbatlas.Client, add []ManagedNamespace, projectID, clusterName string) error {
```

Comment:
[q] Could you rename `conn`  and `add`? it is not clear what they are based on their name